packages/next-on-pages/src/buildApplication/packageManagerUtils.ts

```diff
@@ -53,91 +77,204 @@ export async function getCurrentPackageManager(): Promise<PackageManager> {
 			return 'yarn (classic)';
 		}
 	}
+
 	return 'npm';
 }
 
-export async function getCurrentPackageExecuter(): Promise<string> {
+/**
+ * Retrieves information about the package manager, such as the commands that may be used to run it
+ * or to execute packages.
+ *
+ * @param pm Package manager name.
+ * @returns Information about the package manager.
+ */
+export async function getPackageManagerInfo(
+	pm?: PackageManager,
+): Promise<PackageManagerInfo> {
+	pm ??= await getCurrentPackageManager();
 	const cmd = isWindows() ? '.cmd' : '';
-	const packageManager = await getCurrentPackageManager();
-	switch (packageManager) {
-		case 'npm':
-			return `npx${cmd}`;
+
+	switch (pm) {
+		case 'bun':
+			return {
+				pm,
+				baseCmd: `bun${cmd}`,
+				execArgs: ['x'],
+				infoArgs: ['pm', 'ls'],
+				infoRegex: (name: string) => new RegExp(`^.+ ${name}@(.*)$`, 'im'),
```

Comment:
this doesn't look too clear to me, could we rename it to `getPackageVersionRegex` or something like that? (if I'm understanding correctly that's here for)